### PR TITLE
[SPEC] Inverse operation

### PR DESCRIPTION
**Pull request by PiotrKrzem**
### Details:
 - Add Inverse operation specification

### Tickets:
 - 131139

### CODE REVIEW

File: docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        * ``true`` - output adjugate matrix.
+        * ``false`` - output inverse matrix.
+
+    * **Type**: `bool`
+    * **Default value**: `false`
+    * **Required**: *No*
+
+**Types**
+
+* **T_IN**: any supported floating-point type.
```
Comment (rkazants) [inline docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst:60]: `T_IN` means sort of integer type. It is better to use `T` to be aligned with other specification

File: docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+.. meta::
+  :description: Learn about Inverse-14 - a matrix operation that computes the inverse or adjoint of one or multiple input matrices.
+
+**Versioned name**: *Inverse-14*
+
+**Category**: *Matrix*
+
+**Short description**: *Inverse* operation computes either the inverse or adjoint (conjugate transposes) of one or more square invertible matrices.
+
+**Detailed description**: *Inverse* operation computes the inverse of a square matrix. The operation uses LU decomposition with partial pivoting to compute the inverses.
```
Comment (nshchego) [inline docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst:16]: Why LU decomposition algorithm was chosen? Are there real models with Invert operations and from which framework are they taken? Can it affect accuracy of the PyTorch models?

File: docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            </port>
+        </input>
+        <output>
+            <port id="1" precision="FP32" names="Inverse:0">
+                <dim>2</dim> <!-- batch size of 2 -->
+                <dim>3</dim> <!-- matrix size of 3x3 -->
+                <dim>3</dim>
+            </port>
+        </output>
+    </layer>
```
Comment (nshchego) [inline docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst:83]: Please add example for `adjoint == true`

File: docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+\[ adj(A) = det(A) \cdot A^{-1} \]
+
+where \( A^{-1} \) is the matrix inverse of A, and \( det(A) \) is the determinant of A.
+
+The adjugate matrix exists if and only if the inverse matrix exists.
+
+**Inputs**:
+
+* **1**: `input` - A tensor of shape [..., M, M] and type `T_IN` representing the input square matrices. The inner-most 2 dimensions form square matrices and must be of the same size. **Required.**
```
Comment (nshchego) [inline docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst:38]: `shape [n, row, column]`

Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst:38]: I can adjust it to be `N1, N2, ..., Nn, ROW, COLUMN] - based on the 2 frameworks, the expected input shape has unlimited batch dimensions, followed by 2 matrix dimensions

File: docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst (hunk 2 above)
Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst:16]: It guarantees some degree of numerical stability while keeping high performance, It is used by both [Tensorflow ](https://www.tensorflow.org/api_docs/python/tf/linalg/inv
)and 
[PyTorch](https://pytorch.org/docs/stable/generated/torch.linalg.inv.html#torch.linalg.inv)

File: docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst (hunk 4 above)
Comment (nshchego) [reply docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst:38]: Ok, let's do so

File: docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst (hunk 3 above)
Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst:83]: Added!

File: docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst (hunk 4 above)
Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino_ir/operation_sets/operations_specifications/matrix/Inverse_14.rst]: Applied!

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+  * **Description**: Modifies the return value of the operation. If true, the operation returns the adjoint (conjugate transpose) of the input matrices instead of finding the inverse.
+  * **Range of values**: `true`, `false` 
+
+    * ``true`` - output adjugate matrix.
+    * ``false`` - output inverse matrix. 
+
+  * **Type**: `bool`
+  * **Default value**: `false`
+  * **Required**: *No*
```
Comment (rkazants) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst:60]: please put attribute section above inputs as we do for other ops

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+.. math::
+
+   adj(A) = det(A) \cdot A^{-1}
+
+where **A^{-1}** is the matrix inverse of A, and **det(A)** is the determinant of A.
+
+The adjugate matrix exists if and only if the inverse matrix exists.
+
+**Inputs**:
```
Comment (rkazants) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst:40]: since only one input
```suggestion
**Input**:
```

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+* **1**: `input` - A tensor of shape [B1, B2, ..., Bn, ROW, COL] and type `T` representing the input square matrices. The inner-most 2 dimensions form square matrices and must be of the same size. **Required.**
+
+**Outputs**:
```
Comment (rkazants) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst:44]:
```suggestion
**Output**:
```

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+* **1**: `output` - A tensor with the same type `T` as the input and same shape [B1, B2, ..., Bn, ROW, COL] as the input, representing the inverse matrices (or adjugate matrices) of the input matrices.
+
+**Attributes**:
```
Comment (rkazants) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst:48]:
```suggestion
**Attribute**:
```

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst [hunk 9]
```diff
@@ -0,0 +1,135 @@
+.. {#openvino_docs_ops_matrix_Inverse_14}
+
+Inverse
+=======
+
+
+.. meta::
+  :description: Learn about Inverse-14 - a matrix operation that computes the inverse or adjoint of one or multiple input matrices.
```
Comment (rkazants) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst:8]:
```suggestion
  :description: Learn about Inverse-14 - a matrix operation that computes the inverse or adjoint for a batch of input matrices.
```

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+**Versioned name**: *Inverse-14*
+
+**Category**: *Matrix*
+
+**Short description**: *Inverse* operation computes either the inverse or adjoint (conjugate transposes) of one or more square invertible matrices.
```
Comment (rkazants) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst:14]: also better to say about batch of matrices

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+   adj(A) = det(A) \cdot A^{-1}
+
+where **A^{-1}** is the matrix inverse of A, and **det(A)** is the determinant of A.
+
+The adjugate matrix exists if and only if the inverse matrix exists.
+
+**Inputs**:
+
+* **1**: `input` - A tensor of shape [B1, B2, ..., Bn, ROW, COL] and type `T` representing the input square matrices. The inner-most 2 dimensions form square matrices and must be of the same size. **Required.**
```
Comment (rkazants) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst:42]: `B1, B2, ..., Bn` are the batch dimensions

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+        <input>
+            <port id="0" precision="FP32">
+                <dim>3</dim> <!-- matrix size of 3x3 -->
+                <dim>3</dim> 
+            </port>
+        </input>
+        <output>
+            <port id="1" precision="FP32" names="Inverse:0">
+                <dim>3</dim> <!-- matrix size of 3x3 -->
+                <dim>3</dim>
```
Comment (mitruska) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst:82]: As the two last dimensions are specified as [ROW, COL],
I would suggest to reflect that in the examples like:
```suggestion
                <dim>3</dim> <!-- rows of square matrix -->
                <dim>3</dim> <!-- columns of square matrix -->
```

**Review by mitruska — APPROVED** [review]

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+.. meta::
+  :description: Learn about Inverse-14 - a matrix operation that computes the inverse or adjoint of one or multiple input matrices.
+
+**Versioned name**: *Inverse-14*
+
+**Category**: *Matrix*
+
+**Short description**: *Inverse* operation computes either the inverse or adjoint (conjugate transposes) of one or more square invertible matrices.
+
+**Detailed description**: *Inverse* operation computes the inverse of a square matrix. The operation uses LU decomposition with partial pivoting to compute the inverses.
```
Comment (rkazants) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst:16]: Can we describe here a reference algorithm for LU decomposition? Or at least leave a link to it?

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst (hunk 12 above)
Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst]: Applied!

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst (hunk 11 above)
Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst]: Described!

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst (hunk 10 above)
Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst]: Added!

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst (hunk 9 above)
Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst]: Applied!

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst (hunk 8 above)
Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst]: Applied!

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst (hunk 7 above)
Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst]: Applied!

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst (hunk 6 above)
Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst]: Applied!

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst (hunk 5 above)
Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst]: Applied!

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst (hunk 13 above)
Comment (PiotrKrzem) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/matrix/Inverse_14.rst]: For now I have added the link to the algorithm description, I will add the full algorithm pseudo code after CF!

**Review by rkazants — APPROVED** [review]
